force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -162,10 +175,16 @@ public with sharing class ClassesObjectsMethods {
      * @return A integer between 50 and 100.
      */
     public static Integer generateRandomNumber() {
-        // Use Math class to generate number between 50 and 100
-
-        // Return the random integer
-        return null; // Replace null with the variable you used to store the result
+        Double randomNumber = Math.random();
+        randomNumber = randomNumber * 100;
+        if (randomNumber < 50) {
+            randomNumber += 50;
+        } else if (randomNumber > 100) {
+            randomNumber -= 50;
+        }
+        randomNumber = Math.round(randomNumber);
+        Integer randomInt = (Integer)randomNumber;
+        return randomInt;
```

Comment:
It's possible to do this in one or two lines, but can be difficult. I think this approach is a lot of code and could be simpler.